ncat/ncat_listen.c

```diff
@@ -621,6 +627,65 @@ int read_stdin(void)
     return nbytes;
 }
 
+/* Read from stdin and broadcast to all client sockets. Return the number of
+   bytes read, or -1 on error. */
+int read_stdin_delimiter(void)
+{
+    char buf[DEFAULT_TCP_BUF_LEN];
+
+    /* Converting the ascii valued delimiter parameter to character. */
+    char delimiter = (int)o.delimiter;
+
+    char tc[1];       /* Temporary character */
```

Comment:
You are reading 1 byte at a time, which will kill performance. Better to do some sort of buffered approach where you read in a good amount then split it on the delimiter and send; if there is leftover, move it to the front of the buffer and read again up to the size of the buffer.